pkg/clusterapi/workers_test.go

```diff
@@ -62,6 +62,24 @@ func TestWorkersUpdateImmutableObjectNamesSuccess(t *testing.T) {
 	).To(Succeed())
 }
 
+func TestWorkerObjects(t *testing.T) {
+	g := NewWithT(t)
+	group1 := dockerGroup{
+		MachineDeployment: machineDeployment(),
+	}
+	group2 := dockerGroup{
+		MachineDeployment: machineDeployment(),
+	}
+
+	workers := &dockerWorkers{
+		Groups: []dockerGroup{group1, group2},
+	}
+
+	objects := workers.WorkerObjects()
+
+	g.Expect(len(objects)).To(Equal(6))
```

Comment:
could we verify the content of the slice by comparing the objects?
You can use `ConsistOf` for that